astrodemo.yaml

```diff
@@ -1,252 +1,132 @@
-      mass:
-        $ref: "link here"
-      cost :
-        $ref: "link here"
-      minWorkingTemp:
-        $ref: "link here"
-      maxWorkingTemp:
-        $ref: "link here"
-
-
-
-
-
-
-
-
-
+    communication:
+        type: object
+        properties:
+            power: {$ref: }
+            mass: {$ref: 'https://github.com/chronos-pramantha/RDFvocab/blob/master/ld%2Bjson/SubSystems.json'}
+            cost: {$ref: 'https://github.com/chronos-pramantha/RDFvocab/blob/master/ld%2Bjson/SubSystems.json'}
+            minWorkingTemp: {$ref: 'https://github.com/chronos-pramantha/RDFvocab/blob/master/ld%2Bjson/SubSystems.json'}
+            maxWorkingTemp: {$ref: 'https://github.com/chronos-pramantha/RDFvocab/blob/master/ld%2Bjson/SubSystems.json'}
+    propulsion:
+        type: object
+        properties:
+            power: {$ref: 'https://github.com/chronos-pramantha/RDFvocab/blob/master/ld%2Bjson/SubSystems.json'}
+            mass: {$ref: 'https://github.com/chronos-pramantha/RDFvocab/blob/master/ld%2Bjson/SubSystems.json'}
+            cost: {$ref: 'https://github.com/chronos-pramantha/RDFvocab/blob/master/ld%2Bjson/SubSystems.json'}
+            minWorkingTemp: {$ref: 'https://github.com/chronos-pramantha/RDFvocab/blob/master/ld%2Bjson/SubSystems.json'}
+            maxWorkingTemp: {$ref: 'https://github.com/chronos-pramantha/RDFvocab/blob/master/ld%2Bjson/SubSystems.json'}
+    detector:
+        type: object
+        properties:
+            power: {$ref: 'https://github.com/chronos-pramantha/RDFvocab/blob/master/ld%2Bjson/SubSystems.json'}
+            mass: {$ref: 'https://github.com/chronos-pramantha/RDFvocab/blob/master/ld%2Bjson/SubSystems.json'}
+            cost: {$ref: 'https://github.com/chronos-pramantha/RDFvocab/blob/master/ld%2Bjson/SubSystems.json'}
+            minWorkingTemp: {$ref: 'https://github.com/chronos-pramantha/RDFvocab/blob/master/ld%2Bjson/SubSystems.json'}
+            maxWorkingTemp: {$ref: 'https://github.com/chronos-pramantha/RDFvocab/blob/master/ld%2Bjson/SubSystems.json'}
+    primaryPower:
+        type: object
+        properties:
+            power: {$ref: 'https://github.com/chronos-pramantha/RDFvocab/blob/master/ld%2Bjson/SubSystems.json'}
+            mass: {$ref: 'https://github.com/chronos-pramantha/RDFvocab/blob/master/ld%2Bjson/SubSystems.json'}
+            cost: {$ref: 'https://github.com/chronos-pramantha/RDFvocab/blob/master/ld%2Bjson/SubSystems.json'}
+            minWorkingTemp: {$ref: 'https://github.com/chronos-pramantha/RDFvocab/blob/master/ld%2Bjson/SubSystems.json'}
+            maxWorkingTemp: {$ref: 'https://github.com/chronos-pramantha/RDFvocab/blob/master/ld%2Bjson/SubSystems.json'}
+            density: {$ref: 'https://github.com/chronos-pramantha/RDFvocab/blob/master/ld%2Bjson/SubSystems.json'}
+    backupPower:
+        type: object
+        properties:
+            power: {$ref: 'https://github.com/chronos-pramantha/RDFvocab/blob/master/ld%2Bjson/SubSystems.json'}
```

Comment:
Not sure OpenAPI is compatible with JSON-LD references. Have you checked it?

Reply:
it does not give an error , it accepts any URI so i think we can do this .

Reply:
I'm still not sure it'll work. @Mec-iS  please prot your opinion.

Reply:
@xadahiya for hydra we can have any type of reference right ? so we can easily swap this JSON-LD to another .

Reply:
Yeah, this will work for hydra but I don't think it'll work for OpenAPI. See, the `$ref` needs to point at some set of definitions (defined in OpenAPI specs format) but this JSON-LD specification is entirely different and I don't think any OpenAPI can be built this way.
Getting my point?

Reply:
i can change it to json or something open api supports it what i am saying .

Reply:
yeah, that'll work :+1:

Reply:
@xadahiya isnt jsonld same as json ?but i guess the point is using something that the official open api parser can read right ?

Reply:
Yeah, Json-LD is very similar to json basically it's json with semantics references and contexts.

> but i guess the point is using something that the official open api parser can read right ?

Exactly!